### ISSUE TITLE
pin numpy version in run requirements for meta.yml

### ISSUE THREAD

**Issue by pstjohn** [conda-forge/tensorflow-feedstock#119]
Currently, a basic conda-forge environment with python and tensorflow 2.4.1 will pull numpy 1.20.* by default, which isn't  compatible. 

Would be great if you could add numpy=1.19.* to the run requirements for the meta.yml

**Comment by izahn** [comment]
I'm not sure I understand.

> Currently, a basic conda-forge environment with python and tensorflow 2.4.1 will pull numpy 1.20.* by default, which isn't compatible.

Isn't compatible with what?

**Comment by pstjohn** [comment]
with tensorflow 2.4.1. The TF team tends to [pin numpy versions in their pip releases](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/ci_build/install/install_pip_packages_by_version.sh#L32), and the current meta.yml builds against numpy 1.19.*, but doesn't specify that as a run requirement. 

As anecdotal evidence, in the linked PR from my repo, I got a bunch of [numpy warnings / errors without pinning numpy](https://github.com/NREL/nfp/runs/2657681815), but were fixed when pinning numpy to 1.19.*

**Comment by izahn** [comment]
Good point, maybe this does need a numpy matrix. I'm never really sure about that!

**Comment by jakirkham** [comment]
We don't build with NumPy matrices any more (we got rid of these a while back). We just build against the oldest supported NumPy version and let it float above that at runtime. Though this only matters for libraries that link against NumPy

**Comment by pstjohn** [comment]
Floating above 1.19.* might be problematic in this case, since using numpy 1.20 at runtime with tensorflow has some known issues tensorflow/tensorflow#47691

**Comment by jakirkham** [comment]
Again that comment only pertains to *linking* NumPy. Idk how TensorFlow uses NumPy so can't say whether that is relevant.

**Comment by pstjohn** [comment]
Ah, ok. I suppose to be concrete, I propose changing 
https://github.com/conda-forge/tensorflow-feedstock/blob/f18fb006b85678a5180336e5f99f68a9ba20d5ea/recipe/meta.yaml#L156
to `numpy =1.19`

**Comment by miketrumpis** [comment]
+1 on this proposal. I just ran into this issue last week. Conda allowed numpy 1.20 to install alongside tensorflow 2.4.1. 

Result was this runtime issue: https://github.com/tensorflow/models/issues/9706 (similar to those referenced in https://github.com/tensorflow/tensorflow/issues/47691)

**Comment by h-vetinari** [comment]
The most recent tensorflow builds have all pinned numpy to 1.19 (see also #151 and the linked discussions about numpy compat). Thankfully, TF 2.7. should alleviate this.

In the meantime, is this still an issue for any conda-forge builds in the wild?

**Comment by miketrumpis** [comment]
fyi I just had another opportunity to set up a conda-forge tensorflow environment, which now results in a non-conflicting numpy version. Issue resolved here.

```
numpy                     1.19.5           py38h9894fe3_2    conda-forge
tensorflow                2.6.2           cuda112py38ha230376_1    conda-forge
```

**Comment by ngam** [comment]
Now the pinning seems to be >=1.20
